src/config/strgen.rs

```diff
@@ -277,6 +304,33 @@ mod tests {
     fn basic_match() {
         results_in(
             // Equivalent to `d{ incorrect-os: g, default: e }f`.
+            Spec {
+                string: Some("d".to_owned()),
+                spectype: SpecType::match_expr(
+                    vec![
+                        (
+                            Expr {
+                                exprtype: if cfg!(windows) {
+                                    ExprType::Linux
+                                } else {
+                                    ExprType::Windows
+                                },
+                            },
+                            Spec::from("g"),
+                        ),
+                        (ExprType::Any.into(), Spec::from("e")),
+                    ],
+                    Some(Spec::from("f")),
+                ),
+            },
+            vec!["def"],
+        )
+    }
+
+    #[test]
+    fn match_that_cannot_resolve() {
```

Comment:
Probably better to rename function `match_that_cannot_resolve()` to `unresolvable_match()`.